File: main/contrib/Windows API Code Pack 1.1/source/Samples/PowerMgmtDemo/VB/PowerMgmtDemo/YesNoConverter.vb
'Copyright (c) Microsoft Corporation.  All rights reserved.


Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Data

Namespace Microsoft.WindowsAPICodePack.Samples.PowerMgmtDemoApp
	<ValueConversion(GetType(Boolean), GetType(String))> _
	Public Class YesNoConverter
		Implements IValueConverter
		#Region "IValueConverter Members"

		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
			If targetType IsNot GetType(String) Then
				Return Nothing
			End If

			Dim what As Boolean = CBool(value)
			If what = True Then
				Return "Yes"
			Else
				Return "No"
			End If
		End Function

		' We only support one way binding
		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function

		#End Region
	End Class
End Namespace

File: main/contrib/Windows API Code Pack 1.1/source/Samples/PowerMgmtDemo/CS/PowerMgmtDemo/YesNoConverter.cs
//Copyright (c) Microsoft Corporation.  All rights reserved.

using System;
using System.Windows.Data;

namespace Microsoft.WindowsAPICodePack.Samples.PowerMgmtDemoApp
{
    [ValueConversion(typeof(bool), typeof(string))]
    public class YesNoConverter :IValueConverter
    {
        #region IValueConverter Members

        public object Convert(
            object value, 
            Type targetType, 
            object parameter, 
            System.Globalization.CultureInfo culture)
        {
            if (targetType != typeof(string)) 
                return null;

            bool what = (bool) value;
            if (what == true)
                return "Yes";
            else
                return "No";
        }

        // We only support one way binding
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            throw new NotImplementedException();
        }

        #endregion
    }
}
